src/Http/Http.Abstractions/src/IResultExecutor.cs

```diff
@@ -0,0 +1,25 @@
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http.Abstractions
+{
+    /// <summary>
+    /// Defines an interface for a service which can execute a particular kind of <see cref="IResult"/> by
+    /// manipulating the <see cref="HttpResponse"/>.
+    /// </summary>
+    /// <typeparam name="TResult">The type of <see cref="IResult"/>.</typeparam>
+    /// <remarks>
+    /// Implementations of <see cref="IResultExecutor{TResult}"/> are typically called by the
+    /// <see cref="IResult.ExecuteAsync(HttpContext)"/> method of the corresponding action result type.
+    /// Implementations should be registered as singleton services.
+    /// </remarks>
+    public interface IResultExecutor<in TResult> where TResult : notnull, IResult
+    {
+        /// <summary>
+        /// Asynchronously executes the action result, by modifying the <see cref="HttpResponse"/>.
+        /// </summary>
+        /// <param name="context">The <see cref="HttpContext"/> associated with the current request."/></param>
+        /// <param name="result">The action result to execute.</param>
+        /// <returns>A <see cref="Task"/> which represents the asynchronous operation.</returns>
+        Task ExecuteAsync(HttpContext context, TResult result);
```

Comment:
Thanks for submitting a PR. While having a parallel `IResultExecutor<TResult>` is an interesting idea, this goes a little above-and-beyond what I was thinking about for #32565.

I would look to JsonResult as an example:

https://github.com/dotnet/aspnetcore/blob/302246df4194a779fd526818a09976b7ed739d18/src/Mvc/Mvc.Core/src/JsonResult.cs#L72-L94

It bypasses the `IActionResultExecutor<JsonResult>` and does the default thing when called via `IResult.ExecuteAsync`.

Reply:
Got it, so I'm probably wrong but for this case `RedirectResult` I believe it should have almost the same implementation as:

https://github.com/dotnet/aspnetcore/blob/3d0f22b8b9df34c6b53976563fc12c6267118c75/src/Mvc/Mvc.Core/src/Infrastructure/RedirectResultExecutor.cs#L44-L78

is that correct?

Reply:
Thanks, I got my response from #32679, I feel we are repeating code here, but I am also aware the slight differences may will make a bit hard to reuse existing code so I will follow the pattern done on #32679 and bypass other implementations.